megamek/src/megamek/client/ui/swing/lobby/LobbyActions.java

```diff
@@ -0,0 +1,1312 @@
+        }
+        if (!areForcesEditable(forceList)) {
+            LobbyErrors.showCannotConfigEnemies(frame());
+            return;
+        }
+        client().sendAssignForceFull(forceList, newOwnerId);
+    }
+    
+    void unloadFromBay(Collection<Entity> entities, int bayId) {
+        if (entities.size() != 1) {
+            LobbyErrors.showSingleUnit(frame(), "offload from bays");
+            return;
+        }
+        Entity carrier = CollectionUtil.randomElement(entities);
+        if (!validateUpdate(Arrays.asList(carrier))) {
+            return;
+        }
+        Bay bay = carrier.getBayById(bayId);
+        if (bay == null) {
+            LobbyErrors.showNoSuchBay(frame());
+            return;
+        }
+        Set<Entity> updateCandidates = new HashSet<>();
+        for (Entity loadee : bay.getLoadedUnits()) {
+            lobby.disembark(loadee, updateCandidates);
+        }
+        sendUpdates(updateCandidates);
+    }
+    
+    /**
+     * Creates a fighter squadron from the given list of entities.
+     * Checks if all entities are fighters and if the number of entities
+     * does not exceed squadron capacity. Asks for a squadron name.
+     */
+    void createSquadron(Collection<Entity> entities) {
+        if (!validateUpdate(entities)) {
+            return;
+        }
+        if (entities.stream().anyMatch(e -> !e.isFighter() || e instanceof FighterSquadron)) {
+            LobbyErrors.showOnlyFighter(frame());
+            return;
+        }
+        if (!areAllied(entities)) {
+            LobbyErrors.showLoadOnlyAllied(frame());
+            return;
+        }
+        boolean largeSquadrons = game().getOptions().booleanOption(OptionsConstants.ADVAERORULES_ALLOW_LARGE_SQUADRONS);
+        if ((!largeSquadrons && entities.size() > FighterSquadron.MAX_SIZE) 
+                || entities.size() > FighterSquadron.ALTERNATE_MAX_SIZE) {
+            LobbyErrors.showSquadronTooMany(frame());
+        }
+        
+        // Ask for a squadron name
+        String name = JOptionPane.showInputDialog(frame(), "Choose a squadron designation");
+        if ((name == null) || (name.trim().length() == 0)) {
+            return;
+        }
+        
+        // Now, actually create the squadron
+        FighterSquadron fs = new FighterSquadron(name);
+        fs.setOwner(createSquadronOwner(entities));
+        List<Integer> fighterIds = new ArrayList<>(entities.stream().map(e -> e.getId()).collect(toList()));
+        correctSender(fs).sendAddSquadron(fs, fighterIds);
+    }
+    
+    /** 
+     * Returns a likely owner client; if any of the fighter belongs to the local
+     * player, returns the local player. If not, returns a local bot if any of the
+     * fighters belongs to that; finally, returns the owner of a random one of the 
+     * fighters.
+     */
+    private IPlayer createSquadronOwner(Collection<Entity> entities) {
+        if (entities.stream().anyMatch(e -> e.getOwner().equals(localPlayer()))) {

```

Comment:
Gotta ask, is there a reason we're guessing who owns the squadron that a given entity is in? Do we allow entities from multiple players to form squadrons or something?

Reply:
Yes, loading/squadron/C3/forces are allowed within teams. I didn't want to add an owner dropdown or something for a fighter squadron. Most of the time there'll only be one owner anyway.